internal/command/jsonformat/README.md

```diff
@@ -0,0 +1,231 @@
+for a relevant index (`int` for slice, `string` for map). These functions build
+an entirely populated `Value` object, and the package will then recursively 
+compute the change for the child (and all other children). When a complex change
+has all the children changes, it then passes that into the relevant complex 
+change type.
+
+#### The `change` package
+
+A `Change` should contain all the relevant information it needs to render 
+itself.
+
+The `Change` itself contains the action (eg. `Create`, `Delete`, `Update`), and
+whether this change is causing the overall resource to be replaced (read from 
+the `ReplacePaths` field discussed in the previous section). The actual content 
+of the changes is passed directly into the internal renderer field. The internal
+renderer is then an implementation that knows the actual content of the changes
+and what they represent.
+
+For example to instantiate a change resulting from updating a list of 
+primitives:
+
+```go
+    listChange := change.New(change.List([]change.Change{
+        change.New(change.Primitive(0.0, 0.0, cty.Number), plans.NoOp, false),
+        change.New(change.Primitive(1.0, nil, cty.Number), plans.Delete, false),
+        change.New(change.Primitive(nil, 4.0, cty.Number), plans.Create, false), 
+        change.New(change.Primitive(2.0, 2.0, cty.Number), plans.NoOp, false)
+    }, plans.Update, false))
+```
+
+##### The `Render` function
+
+Currently, there is only one way to render a change, and it is implemented via
+the `Render` function. In the future, there may be additional rendering 
+capabilities, but for now the `Render` function just passes the call directly 
+onto the internal renderer.
+
+Rendering the change with: `listChange.Render(0, RenderOpts{})` would
+produce:
+
+```text
+[
+    0,
+  - 1 -> null,
+  + 4, 
+    2,
+]    
+```
+
+Note, the render function itself doesn't print out metadata about its own change
+(eg. there's no `~` symbol in front of the opening bracket). The expectation is
+that parent changes control how child changes are rendered (so are responsible)
+for deciding on their opening indentation, whether they have a key (as in maps, 
+objects, and blocks), or how the action symbol is displayed.
+
+In the above example, the primitive renderer would print out only `1 -> null` 
+while the surrounding list renderer is providing the indentation, the symbol and
+the line ending commas.
+
+##### Implementing new change types
+
+To implement a new change type, you must implement the internal Renderer 
+functionality. To do this you create a new implementation of the 
+`change/renderer.go`, make sure it accepts all the data you need, and implement
+the `Render` function (and any other additional render functions that may 
+exist). Some changes publish warnings that should be displayed alongside them, 
+if your new change has no warnings you can use the `NoWarningsRenderer` to avoid
```

Comment:
```suggestion
exist).

Some changes publish warnings that should be displayed alongside them. 
If your new change has no warnings you can use the `NoWarningsRenderer` to avoid
```